app/controllers/traveler/journeys_controller.rb

```diff
@@ -26,12 +25,14 @@ def create
   end
 
   def show
-    @journey = Journey.find(params[:id])
+      @journey = Journey.find(params[:id])
   end 
   
   def edit
-    @journey = Journey.find(params[:id])
-  end
+    if current_user.role=='traveler'
+     @journey = Journey.find(params[:id])
```

Comment:
use find instead of find_by @aleemchishti

Reply:
done